client/src/components/Form/Elements/FormText.vue

```diff
@@ -0,0 +1,157 @@
+            <b-col>
+                <component
+                    :is="componentName"
+                    :id="id"
+                    v-model="currentValue"
+                    class="text-input"
+                    :readonly="readonly"
+                    :placeholder="placeholder"
+                    :style="style"
+                    :type="localType"
+                    @change="onInputChange" />
+
+                <datalist v-if="datalist && !inputArea && !multiple" :id="`${id}-datalist`">
+                    <option v-for="data in datalist" :key="data.value" :label="data.label" :value="data.value"></option>
+                </datalist>
+            </b-col>
+        </b-row>
+    </div>
+</template>
+
+<script>
+export default {
+    props: {
+        value: {
+            // String; Array for multiple
+            required: false,
+            default: "",
+        },
+        id: {
+            type: String,
+            required: false,
+        },
+        type: {
+            type: String,
+            required: false,
+            default: "text",
+        },
+        area: {
+            // <textarea> instead of <input> element
+            type: Boolean,
+            required: false,
+            default: false,
+        },
+        multiple: {
+            // Allow multiple entries to be created
+            type: Boolean,
+            required: false,
+            default: false,
+        },
+        readonly: {
+            type: Boolean,
+            required: false,
+            default: false,
+        },
+        placeholder: {
+            type: String,
+            required: false,
+        },
+        color: {
+            type: String,
+            required: false,
+        },
+        styleObj: {
+            // This will be applied to the input element
+            type: Object,
+            required: false,
+            default: () => {},
+        },
+        datalist: {
+            // Display list of suggestions in autocomplete dialog
+            type: Array,
+            required: false,
+        },
+
+        // These last three are for handling special input cases only
+
+        optional: {
+            type: Boolean,
+            default: false,
+            required: false,
+        },
+        model_class: {
+            type: String,
+            required: false,
+        },
+        data: {
+            type: Object, // ?
+            required: false,
+        },
+    },
+    data() {
+        let inputArea;
+        if (["SelectTagParameter", "ColumnListParameter"].includes(this.model_class) || (this.options && this.data)) {
+            inputArea = this.multiple;
+        }
+        return {
+            dismissSecs: 5,
+            dismissCountDown: 0,
+            errorMessage: "",
+            inputArea: inputArea,
+        };
+    },
+    computed: {
+        localType() {
+            // this seems silly, if we're in a component called FormText ... this has to be text??
```

Comment:
I think, we can avoid this and simplify this module by separating the `<components />` in the template above into explicitly into `b-form-textarea` and `b-form-input` instead of calling `componentName` and instead of using a generic component. In that case `inputArea` and `area` could be the same and result consistently in a `b-form-textarea` which can only be `text`.

Reply:
I think inputArea has to remain it's own entity, since it's dealing with some of the special case props.  I've separated out the generic component in the beginning and taking the logic from componentName to determine when to build one over another.

Reply:
Thanks for fixing these issues, I made an attempt to consolidate the two.